avm/res/kusto/cluster/main.bicep

```diff
@@ -440,17 +440,17 @@ type diagnosticSettingType = {
 }[]?
 
 type customerManagedKeyType = {
-  @description('Required. The resource ID of a key vault to reference a customer managed key for encryption from.')
-  keyVaultResourceId: string
-
   @description('Required. The name of the customer managed key to use for encryption.')
   keyName: string
 
+  @description('Required. The resource ID of a key vault to reference a customer managed key for encryption from.')
+  keyVaultUri: string
```

Comment:
Change should be undone as it deviates from the corresponding [spec](https://azure.github.io/Azure-Verified-Modules/specs/shared/interfaces/#customer-managed-keys). We opted to fetch the key vault URI for the user if the resource ID is provided (as per the description). Reason being: This allows us to be consistent across AVM modules. 

However, this surfaced an entirely different problem of this module, that is, that the CMK implementation is not following the same standard. That being said - I did already implement a 'fix' in this [PR](https://github.com/Azure/bicep-registry-modules/pull/3766/files#diff-0d5e4ddc60e8fe60ecd0bff75b87c709c22a47082a7a4a3205eb44b75c82e3ab), which is currently just waiting another PR  https://github.com/Azure/bicep-registry-modules/pull/3773 to publish a new version of the CMK type to be implemented. 

If alright with you, I'd ask you to park the changes to CMK implemented in this PR and only change anything but 😉

Reply:
maybe park this PR in general as I can see several other overlapping changes too. Also, following the changes, the workflow will run successfully 💪